src/manifest.c

```diff
@@ -695,12 +695,26 @@ struct manifest *load_mom_err(int version, bool latest, bool mix_exists, int *er
 struct manifest *load_manifest(int current, int version, struct file *file, struct manifest *mom, bool header_only)
 {
 	struct manifest *manifest = NULL;
+	struct file *peer = NULL;
 	int ret = 0;
 	bool retried = false;
 
 retry_load:
 	ret = retrieve_manifests(current, version, file->filename, file, file->is_mix);
 	if (ret != 0) {
+		if (file->is_iterative == 1 && file->full_manifest && !retried) {
+			/* Since the iterative manifest failed to be loaded,
+			 * fall back to using the full manifest */
+			peer = file->peer;
+			file = file->full_manifest;
+			/* re-link the full manifest */
+			file->peer = peer;
```

Comment:
This block got a bit hard to read. I think it's easier to write that this way:

    file->full_manifest->peer = file->peer;
    file = file->full_manifest;

Does it make sense?

Reply:
yep, good idea, looks nicer that way, thanks for the tip